imap_processing/ultra/l1b/cullingmask.py

```diff
@@ -0,0 +1,35 @@
+"""Calculates Culling Mask."""
+
+
+def calculate_cullingmask(extended_spin_dict):
+    """
+    Create dictionary with defined datatype for Culling Mask Data.
+
+    Parameters
+    ----------
+    extended_spin_dict: : dict
+        L1b data dictionary.
+
+    Returns
+    -------
+    cullingmask_dict : dict
+        Dictionary containing the data.
+    """
+    cullingmask_dict = {}
+
+    # Placeholder for bitwise filtering based on quality flags
+    cullingmask_dict["epoch"] = extended_spin_dict["epoch"]
```

Comment:
Would it be reasonable to make these dictionaries dataclasses instead? You could subclass to include different or extended data, or to use them for different cases, or even just to distinguish them. Basically make all these keys attributes in the dataclass?

 I know this is just a placeholder so it's possible that suggestion is overkill, but I find dataclasses provide some abstraction so I can easily go back in and add logic. For example, I could just start by initializing the dataclass and moving that around, but then later, add in logic to cull bad datapoints, without needing to update everywhere that the dataclass is used.

Reply:
I love this idea. Let's discuss. Do you plan on using it for MAG? I always like seeing examples to get ideas.

Reply:
Yup, definitely happy to talk about it. I am actually doing something very interesting for GLOWS that would line up quite well with what you have here - if we plan to meet end of day today or tomorrow, it should be complete enough for you to get a full idea of how it works!